__tests__/reporters/json.test.ts

```diff
@@ -268,49 +272,83 @@ describe('json reporter', () => {
     expect(screenshot1).toEqual(screenshot2);
   });
 
-  it('write screenshot blob data', async () => {
+  describe('screenshots', () => {
     const sourceDir = join(FIXTURES_DIR, 'screenshots');
     const destDir = join(helpers.CACHE_PATH, 'screenshots');
-    mkdirSync(destDir, { recursive: true });
-    fs.copyFileSync(
-      join(sourceDir, 'content.json'),
-      join(destDir, 'content.json')
-    );
-    runner.emit('journey:end', {
-      journey: j1,
-      start: 0,
-      status: 'failed',
-      ssblocks: false,
+    beforeAll(() => {
+      mkdirSync(destDir, { recursive: true });
+      fs.copyFileSync(
+        join(sourceDir, 'content.json'),
+        join(destDir, 'content.json')
+      );
     });
-    const stepEnd = (await readAndCloseStreamJson()).find(
-      json => json.type == 'step/screenshot'
-    );
-    expect(stepEnd).toMatchObject({
-      step: {
-        name: 'launch app',
-        index: 1,
-      },
-      blob: expect.any(String),
-      blob_mime: 'image/jpeg',
+
+    afterAll(() => {
+      fs.rmdirSync(destDir, { recursive: true });
     });
-    fs.rmdirSync(destDir, { recursive: true });
-  });
 
-  it('write screenshot block & reference docs', async () => {
-    const sourceDir = join(FIXTURES_DIR, 'screenshots');
-    const destDir = join(helpers.CACHE_PATH, 'screenshots');
-    mkdirSync(destDir, { recursive: true });
-    fs.copyFileSync(
-      join(sourceDir, 'content.json'),
-      join(destDir, 'content.json')
-    );
-    runner.emit('journey:end', {
-      journey: j1,
-      start: 0,
-      status: 'failed',
-      ssblocks: true,
+    const emitEnd = (options, status = 'failed' as StatusValue) =>
+      runner.emit('journey:end', {
+        journey: j1,
+        start: 0,
+        status,
+        options,
+      });
+
+    it('write whole blobs data ', async () => {
+      emitEnd({
+        screenshots: 'on',
+        ssblocks: false,
+      });
+      const stepEnd = (await readAndCloseStreamJson()).find(
```

Comment:
Can we rename these `stepEnd` variables? These look to be copy/pasted from earlier tests where they were indeed `step/end` docs, but they now should be named `stepScreenshot` or `screenshot` etc.

Reply:
Good catch, Fixed!